src/main/java/edu/kis/powp/jobs2d/visitor/TransformCommandVisitor.java

```diff
@@ -0,0 +1,27 @@
+package edu.kis.powp.jobs2d.visitor;
+
+import edu.kis.powp.jobs2d.command.*;
+
+import java.sql.Driver;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public interface TransformCommandVisitor {
+
+    SetPositionCommand visit (SetPositionCommand command);
+
+    OperateToCommand visit(OperateToCommand command);
+
+    default CompoundCommand visit(ICompoundCommand commands) {
+        List<DriverCommand> modifiedCommands = new ArrayList<>();
+        Iterator<DriverCommand> iterator = commands.iterator();
+        while(iterator.hasNext()) {
+            DriverCommand command = iterator.next();
+            command = command.accept(this);
```

Comment:
instead of returning, please add a field to TransformCommandVisitor so you can get a command generated by the last accept call.